tensorboard/components/experimental/plugin_util/core-host-impl.ts

```diff
@@ -0,0 +1,29 @@
+/* Copyright 2020 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+/**
+ * Implements core plugin APIs.
+ */
+tb_plugin.host.listen('experimental.GetURLPluginData', (context) => {
+  const prefix = `p.${context.pluginName}.`;
```

Comment:
Good point, I didn't consider that.  @wchargin wdyt about disallowing periods?
I've not found any plugins that use periods, but technically it would be a breaking change.
Alternatively, we could use some character like `~` or `*`, and just communicate that in the docs.

https://github.com/tensorflow/tensorboard/pull/3160

Reply:
Disallowing periods seems reasonable to me. It prevents putting (some
forms of) version numbers in plugin names, but that seems fine. We could
use `~`, but I think that that would be confusing (it also frequently
breaks links in PDFs written by TeX novices). We can’t use `*` without
encoding it, because it is not an unreserved character per [RFC 3986,
section 2.3][rfc]. So given that we want to permit `[A-Za-z0-9_-]`,
reserving `.` seems like the best option.

[rfc]: https://tools.ietf.org/html/rfc3986#section-2.3